### PR TITLE
Implement Replit Unified Embed

### PR DESCRIPTION
**Pull request by msarit**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Replit embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15570

## QA Instructions, Screenshots, Recordings

Example Replit URLs:
https://replit.com/@msarit/Shell-Challenge#index.html
https://replit.com/@msarit/Shell-Challenge

Replit Inputs:
@msarit/Shell-Challenge#index.html
@msarit/Shell-Challenge

- As a user, create a Replit liquid tag using this format: `{% embed <replit-url> %}`. Please use the example URLs provided above. The Replit Liquid Tag should render properly.
- Also create Replit liquid tag using the old method: `{% replit <replit-input> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
None

## Added/updated tests?
- [X] Yes

## [Forem core team only] How will this change be communicated?
- [X] I will share this change internally with the appropriate teams
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None

## How this PR makes me feel:
Ratatouille cooking

### CODE REVIEW

**Comment by djuber** [comment]
Looks like your branch is older than the spotify changes,  you might need to merge/rebase master to resolve the conflicts in the registry spec?

**Comment by msarit** [comment]
> Looks like your branch is older than the spotify changes, you might need to merge/rebase main to resolve the conflicts in the registry spec?

Doing so now 😅

**Comment by djuber** [comment]
local testing - for the embed tag only the url format is accepted (trying to `{% embed @msarit/Shell-Challenge %}` gives a local link to `/${user}/@msarit/Shell-Challenge` which is a 404 - should we be rejecting that instead of building a link or is embed going to default to assuming its content is a local page?

for the legacy `replit` liquid tag both formats are accepted.

https://gist.github.com/djuber/b9372494b0f64b5f5e3203e8fe2a5ec3

http://localhost:3000/admin_mcadmin/@msarit/Shell-Challenge seems like something we might not want to have generated as a link given that input to `embed`.

**Review by djuber — APPROVED** [review]
works as advertised - my only question is around the behavior of embed when the format is not recognized (can I pass anything to embed and it defaults to making a local link?)

When I preview these "generic" embeds, like {% embed thing %} I get an absolute link to `/thing`, but when I publish I get `/${user}/thing`.

Out of scope for _this_ change, not sure if that's a known or expected behavior.

File: spec/liquid_tags/replit_tag_spec.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 RSpec.describe ReplitTag, type: :liquid_tag do
   describe "#id" do
     let(:replit_id) { "@WigWog/PositiveFineOpensource" }
+    let(:invalid_replit_id) { "@Wig-Wog/PositiveFineOpensource" }
```
Comment (jgaskins) [inline spec/liquid_tags/replit_tag_spec.rb:6]: It took me a couple minutes to figure out what was different here, like those "spot the differences between these images" games in the children's magazines I used to read when I went to the dentist.

This change should make it a bit more obvious what's going on:

```suggestion
    let(:replit_id) { "@WigWog/PositiveFineOpensource" }
    let(:invalid_replit_id) { "@Cant-Have-Dashes/PositiveFineOpensource" }
```

**Review by jgaskins — APPROVED** [review]
Just a suggestion on clarity. Otherwise, this looks great!

**Comment by msarit** [comment]
> works as advertised - my only question is around the behavior of embed when the format is not recognized (can I pass anything to embed and it defaults to making a local link?)

@djuber The fallback to a link (that most likely leads nowhere) is an intentional behavior for the unified-embed.

However, you have provoked me to think about the user experience of that behavior. Intuitively, I would expect the embed to fail if the url I passed in was invalid, along with an error message. That at least gives me the choice of leaving the embed out altogether, or creating my own link to the resource I attempted to embed.

The unified-embed documentation (to be written after I have implemented all unified embeds) will clearly explain the expected use of `{% embed url %}`. But I do want to think about the embed unhappy-path a bit more.

Thanks for provoking these thoughts!

**Pull request merged**